src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs

```diff
@@ -54,60 +46,63 @@ private bool IsValidScanTarget(string file)
                    MachOBinary.CanLoadBinary(uri);
         }
 
-        protected override BinaryAnalyzerContext CreateContext(AnalyzeOptions options, IAnalysisLogger logger, RuntimeConditions runtimeErrors, PropertiesDictionary policy = null, string filePath = null)
+        public override BinaryAnalyzerContext InitializeContextFromOptions(AnalyzeOptions options, ref BinaryAnalyzerContext context)
         {
-            if (logger is AggregatingLogger aggregatingLogger && this.Telemetry?.TelemetryClient != null)
+            if (this.Telemetry != null)
```

Comment:
>if (this.Telemetry != null) [](http://example.com/codeflow?start=12&length=27)

this should be changed back to at least check 
`this.Telemetry?.TelemetryClient != null`

Reply:
@shaopeng-gh But in that case, we would never end up creating an aggregating logger either. I see that was the situation before. For my understanding: if the TelemetryClient == null, we do not perform any logging? Is there a default logger we can pass to the aggregating logger if the telemetry client is null?

Reply:
The safest thing to do is put it back in the condition it was). I'm not sure we have a test that exercises this code directly from the options (our new pipeline model will allow us to write this). Also, we do not need to populate the logger if null here, the calling code will do that. The convention is, if the logger is null and you need to add one, creating an aggregating logger and add yours to it.

If the logger is non-null, in most cases it s/be an aggregating logger, just add yours.

There is one exception, there is a narrow window for console reporting where we don't yet know the reporting levels on the options/config.

Reply:
When I ran the code, the `this.Telemetry` object is not null but the `this.Telemetry.TelemetryClient` property is null, the line below throw null ref error.

Reply:
This is my only comment left